### DISCUSSION TITLE
How to config local dependency for filament source code?

### DISCUSSION THREAD

**Discussion by kirinzer** [google/filament#3860] [Q&A]
I am trying to discover the workflow of filament, and I find a podspec file in the repo. Unfortunately, the dependency in that file is also static lib. But I want to have a podspc file that can dependence the source code. Is there anybody who could tell me how to write a podspec file? Actually, I use Xcode to run all samples. Maybe a cmake file could do the same thing, if you know where it is and how to use it, please tell me, thanks so much!!!

**Comment by romainguy** [comment]
@bejado podspecs are for iOS right?

**Reply by bejado** [reply]
Correct, the Podspec in our repo links your iOS app against our pre-built static libs. Building Filament from source is more complicated than could be expressed in a Podspec. You can try building Filament through Xcode by passing `Xcode` as the generator to CMake:

```
cmake -G Xcode .. -T buildsystem=1
```

This isn't regularly tested though and I'm not sure how well it will work.

**Reply by kirinzer** [reply]
I had tried your solution. After running that command I got a TNT.project, and I select a target "hello triangle". But it can not build successfully. Some files can not be found like this:
```
#include "generated/resources/resources.h"
```
I try to search it in the root path.  Add resources folder into TNT.project. But it still does not work.
the new issues look like this:
```
Command /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang failed with exit code 1

Command /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang failed with exit code 1

Command /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang failed with exit code 1
```
I guess these files should be generated automatically.

**Reply by bejado** [reply]
Here are the steps I took that worked:

```
cd filament
mkdir build-xcode
cd build-xcode
cmake -G Xcode .. -T buildsystem=1
```

I opened the Xcode proejct, chose the `hellotriangle` target, and hit ⌘B to build.

I'm using CMake version 3.19.5 and Xcode version 12.5.

**Reply by kirinzer** [reply]
All right, the steps are the same with me. A little different is the version about CMake and Xcode.
My CMake version is 3.19.2 and my Xcode is 12.3(12c33).
Anyway, I will try again later. Thanks for your patient :)
